app/models/identity.rb

```diff
@@ -39,6 +39,8 @@ def self.build_from_omniauth(provider)
   end
 
   def email
-    auth_data_dump.info.email
+    return auth_data_dump.info.email if auth_data_dump&.info&.email
+
+    "No email found. Please relink your #{provider} account to avoid errors."
```

Comment:
Another tip here, you can make turn this into a constant to save a bit of memory, like so:
```ruby
# constants are usually defined near the top of the class
class Identity < ApplicationRecord
  NO_EMAIL_MSG = "No email found. Please relink your %<provider> account to avoid errors.".freeze
  
  # ... down below
  
  def email
    return auth_data_dump.info.email if auth_data_dump&.info&.email

    format(MESSAGES[:no_email_found], provider: provider)
  end
end
```

The `format` method will convert the `provider` key value pair just like in normal string interpolation, allowing us to use constants with conditional strings. You can also see `app/models/user.rb` as an example:
https://github.com/forem/forem/blob/0cac61024f63a93da0965a8a202a4cc1952b575d/app/models/user.rb#L76-L82